### PR TITLE
util/codec: fix decimal encoding.

### PR DESCRIPTION
**Pull request by coocood**
The decimal encoding for negative values with the same digit prefix was broken, like -0.12 and -0.1234, the result should be  [-0.1234, -0.12"] rather than [-0.12, -0.1234].

This PR fixes it by use EncodeBytesDesc and DecodeBytesDesc directly.

### CODE REVIEW

File: util/codec/bytes.go [hunk 1]
```diff
@@ -59,33 +56,64 @@ func EncodeBytes(b []byte, data []byte) []byte {
...
-		result = append(result, marker)
+		if desc {
+			result = append(result, byte(padCount))
+		} else {
+			result = append(result, encMarker-byte(padCount))
+		}
+	}
+	if desc {
+		// We don't want to modify the data, restore to the original.
+		reverseBytes(data)
```
Comment (siddontang) [inline util/codec/bytes.go]: why not reversing the bytes when appending but reversing data twice?

Comment (coocood) [inline util/codec/bytes.go]: Then we have to do reverse for every group.

Comment (siddontang) [inline util/codec/bytes.go]: I don't think it is hard to do it, you can add a helper function to do it.

Comment (siddontang) [inline util/codec/bytes.go]: If the data is huge, reversing twice is not efficient.

**Comment by coocood** [comment]
@siddontang @qiuyesuifeng PTAL

**Comment by shenli** [comment]
LGTM

File: util/codec/decimal.go [hunk 2]
```diff
@@ -108,12 +108,15 @@ func EncodeDecimal(b []byte, d mysql.Decimal) []byte {
 	// For negtive exp, do bit reverse for exp.
 	// For negtive decimal, do bit reverse for exp and value.
 	expVal = encodeExp(expVal, expSign, valSign)
-	codecValue(value, valSign)
 
 	b = append(b, byte(valSign))
 	b = append(b, byte(expSign))
 	b = EncodeInt(b, expVal)
-	b = EncodeBytes(b, value)
+	if valSign == negativeSign {
```
Comment (qiuyesuifeng) [inline util/codec/decimal.go]: Seems line 109 can be removed?

Comment (qiuyesuifeng) [inline util/codec/decimal.go]: And also codecValue function.

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**